scrapy/utils/url.py

```diff
@@ -103,3 +103,35 @@ def guess_scheme(url):
         return any_to_uri(url)
     else:
         return add_http_if_no_scheme(url)
+
+
+def strip_url_credentials(url, origin_only=False, keep_fragments=False):
```

Comment:
Could you please add a docstring to this function?

The function is named 'strip credentials', but it also strips port, I think this is not intuitive.